### DISCUSSION TITLE
How to `preventOverflow` in floating-ui, coming from popper.js

### DISCUSSION THREAD

**Discussion by jgodi** [floating-ui/floating-ui#1642] [Q&A] [answered]
Hello, I am trying to move away from our Popper.js usage to the new floating-ui. I cannot find a way to do a replacement for the overflow prevention.

Basically, I want to keep the floating-ui in the viewport as the anchor scrolls outside of the viewport. Before, we could do something like this:

```
preventOverflow: {
    boundariesElement: 'window',
},
```

In my new implementation, I am using:

```
  const {
    x,
    y,
    reference,
    floating,
    placement,
    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
    update,
    refs,
  } = useFloating({
    placement: placementProp,
    middleware: [
      offset(offsetProp),
      flip(),
      shift({ padding: offsetProp }),
      arrow({ element: arrowRef }),
    ],
  });
```

I have tried to pass options to shift for the `detectOverflow`, but cannot find the right combination to accomplish this.

Thank you!

**Comment by atomiks** [comment, marked as answer]
Do you have a CodeSandbox demo for context?

**Reply by jgodi** [reply]
I cannot make one easily. My main question is if there is a way to keep the floating element within the viewport as the user scrolls vs. having the element scroll outside of the viewport.

**Reply by atomiks** [reply]
`shift()` does that by default

**Reply by atomiks** [reply]
Perhaps you're looking for [`autoUpdate`](https://floating-ui.com/docs/autoUpdate) to update the position on scroll. But `shift()` will do what you want without any options needed

**Reply by jgodi** [reply]
Thank you
